### ISSUE TITLE
Show CFM payment data on purchase confirmation. 

### ISSUE THREAD

**Issue by mintplugins** [easydigitaldownloads/edd-checkout-fields#112]
Any data collected through CFM should be output into the Purchase Confirmation. Currently it is not. This allows the customer to see the details of what they just purchased. 

It will also allow Vendors to see that data in FES.

**Show CFM data here:**
<img width="690" alt="screen shot 2017-02-10 at 1 48 05 pm" src="https://cloud.githubusercontent.com/assets/1638128/22839484/9910eefc-ef97-11e6-878c-69778af83a00.png">

**Comment by pippinsplugins** [comment]
Initial WIP PR submitted!

**Comment by brashrebel** [comment]
Looking nice so far. It is throwing array to string conversion notices for all the field types which return an array (file upload, radio, checkbox, multi-select, etc.) A test with only fields returning strings was successful and looks exactly as I'd expect it to.

**Comment by pippinsplugins** [comment]
I'll fix that.

I'm also going to add two settings, one to enable / disable the output and one to control the header text shown.

**Comment by pippinsplugins** [comment]
@brashrebel give that another try?

I also added settings: 
<img width="1130" alt="screen shot 2017-05-09 at 11 00 32 am" src="https://cloud.githubusercontent.com/assets/1034109/25860291/bc672aca-34a6-11e7-9e44-ea7a4a7a158c.png">

**Comment by brashrebel** [comment]
Both settings work for me just fine. When enabled, I see every custom checkout field on every payment, including those which existed before enabling CCFM (is that the desired behavior?). Order of fields is retained which is cool too.

**Comment by pippinsplugins** [comment]
It pulls data for all fields that are currently in the form editor.

**Comment by pippinsplugins** [comment]
Todo: account for checkboxes, radio buttons, and file uploads.

**Comment by chriscct7** [comment]
Not sure why you are using `$field->get_field_value_frontend()`. It's not designed for display, but rather to get the raw value. If you're trying to display the field value in CFM you should use `$field->export_data()` (in FES you should use `$field->formatted_data()` )

**Comment by pippinsplugins** [comment]
Because I didn't know that method existed :)

**Comment by pippinsplugins** [comment]
Final display of data:
<img width="1168" alt="screen shot 2017-05-10 at 11 01 27 am" src="https://cloud.githubusercontent.com/assets/1034109/25908534/0c99f5da-3570-11e7-9bca-5ec6a1025ba1.png">

**Comment by pippinsplugins** [comment]
Merged into release/2.1.

**Comment by mintplugins** [comment]
This looks good to me! 👍

**Comment by mintplugins** [comment]
Could we make these details show on the Vendor's "orders" screen as well? This way Vendors can review custom submitted information.

**Comment by pippinsplugins** [comment]
@mintplugins let's do that later on in a different issue.

**Comment by mintplugins** [comment]
This appears to be broken. 

Before Checkout (fields are filled out):
<img width="449" alt="screen shot 2017-11-12 at 5 27 26 pm" src="https://user-images.githubusercontent.com/1638128/32704172-d8aa59b2-c7ce-11e7-9520-b032021bd6b0.png">

After Checkout (fields are blank):
<img width="494" alt="screen shot 2017-11-12 at 5 27 37 pm" src="https://user-images.githubusercontent.com/1638128/32704175-e5a03a6a-c7ce-11e7-8a5d-67417cc1fc9a.png">

I noticed that the fields which are blank are all user meta fields (not payment meta fields). Possibly related: https://github.com/easydigitaldownloads/edd-checkout-fields/issues/138

Note that the fields did correctly populate the Customer's information (First/Last Name etc):
<img width="879" alt="screen shot 2017-11-12 at 5 33 31 pm" src="https://user-images.githubusercontent.com/1638128/32704215-a52dd9b4-c7cf-11e7-98ba-1372c56e338f.png">


Additionally, I tested without the checkbox field because I know checkboxes are prone to cause strange issues. However, removing the checkout field had no affect so I don't believe it is checkbox related. 

If someone else could test to confirm that would be great. 

Testing Steps:
1. Make sure your settings for the Checkout Fields Manager plugin are set to show the fields on the Purchase Confirmation page. 
2. In your checkout fields manager add First/Last Name fields. Also add an email field, a text field which is saved to the user meta, and a field which is saved to the payment meta. 
3. Purchase the product with all fields filled out. 
4. Check if the purchase confirmation page shows the entered data or if it is blank.
